companion: return 401 for invalid access token

ifedapoolarewaju opened the pull request:
return 401 for invalid access token so that the `/authorized` endpoint can be deprecated

arturi commented:
Authenticated, then tried to use Google Drive again after some time:

<img width="812" alt="Screen Shot 2019-03-16 at 16 30 15" src="https://user-images.githubusercontent.com/1199054/54475965-d8751200-4808-11e9-8935-30dc2a481ff6.png">

```
::1 - - [16/Mar/2019:13:29:47 +0000] "OPTIONS /drive/list/root HTTP/1.1" 200 8 "http://localhost:3452/" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10.14; rv:65.0) Gecko/20100101 Firefox/65.0"
uppy: 2019-03-16T13:29:48.015Z [error] provider.drive.teamDrive.error [object Object]
 { error:
   { errors: [ [Object] ],
     code: 401,
     message: 'Invalid Credentials' } } 
::1 - - [16/Mar/2019:13:29:48 +0000] "GET /drive/list/root HTTP/1.1" 500 199 "http://localhost:3452/" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10.14; rv:65.0) Gecko/20100101 Firefox/65.0"
```

arturi commented:
And when I reload a page and try to open a provider after some time, I get this on the auth page:

<img width="814" alt="Screen Shot 2019-03-18 at 14 45 33" src="https://user-images.githubusercontent.com/1199054/54527971-970d6f80-498c-11e9-9be2-44dd012728ae.png">

So maybe not showing an Informer is a way to go, since we can’t only show it on non-auth pages, right?

ifedapoolarewaju commented:
it shows the informer message and it also switches to the auth page. It will always switch to the auth page once it detects that auth is required at a given point, but also shows informer message while at it. 

Are you saying that we should not show the informer message anymore? But just swtich to the Auth page?

arturi commented:
I’m thinking out loud. It’s correct to show an error when I connect to Google Drive, then wait a while, and my token expires, so when I’m thrown at the Authenticate screen, some explanation would be nice probably. But I saw the error the first time I tried authenticate. So my experience with Uppy and Google Drive starts with an error.

ifedapoolarewaju commented:
@arturi I have fixed the issue with the google auth. I have also changed the informer message for auth errors.

While it was suggested that we should not show informer message on first auth error, I wonder if it's still worth it for me to implement this. The flow seems better now that I have changed the informer message, and I wonder if the need to add the extra complexity of showing informer message only after the second auth error is necessary. cc @goto-bus-stop

Let me know what you think

arturi commented on packages/@uppy/provider-views/src/index.js:
```diff
@@ -534,10 +535,8 @@ module.exports = class ProviderView {
             handleAuth={this.handleAuth}
-            handleDemoAuth={this.handleDemoAuth}
-            checkAuthInProgress={checkAuthInProgress} />
+            handleDemoAuth={this.handleDemoAuth} />
```
Should we remove this, have not been used for a while, right?

ifedapoolarewaju replied:
I guess so, yes

arturi commented:
> While it was suggested that we should not show informer message on first auth error, I wonder if it's still worth it for me to implement this. 

but `preFirstRender` and `onFirstRender` are still introduced?

ifedapoolarewaju commented:
> > While it was suggested that we should not show informer message on first auth error, I wonder if it's still worth it for me to implement this.
> 
> but `preFirstRender` and `onFirstRender` are still introduced?

yes but they serve a different purpose

tkejr commented:
facing this error
